src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2166,195 +2166,53 @@ private ConsumeTokenResult ConsumeNextTokenUntilAfterAllCommentsAreSkippedMultiS
             return ConsumeTokenResult.NotEnoughDataRollBackState;
         }
 
-        private bool SkipCommentMultiSegment()
+        private bool ConsumeCommentMultiSegment()
         {
-            // Create local copy to avoid bounds checks.
-            ReadOnlySpan<byte> localBuffer = _buffer.Slice(_consumed + 1);
-            int leftOver = 2;
-
-            if (localBuffer.Length == 0)
-            {
-                if (IsLastSpan)
-                {
-                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, JsonConstants.Slash);
-                }
-
-                if (!GetNextSpan())
-                {
-                    if (IsLastSpan)
-                    {
-                        ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, JsonConstants.Slash);
-                    }
-                    return false;
-                }
-
-                _totalConsumed++;
-                _bytePositionInLine++;
-                localBuffer = _buffer;
-                leftOver = 1;
-            }
-
-            byte marker = localBuffer[0];
+            long prevTotalConsumed = BytesConsumed;
+            SequencePosition start = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed);
+            bool ret = SkipCommentMultiSegment(out int tailBytesToIgnore);
 
-            if (marker == JsonConstants.Slash)
+            if (ret)
             {
-                return SkipSingleLineCommentMultiSegment(localBuffer.Slice(1), leftOver);
-            }
+                SequencePosition end = new SequencePosition(_currentPosition.GetObject(), _currentPosition.GetInteger() + _consumed);
 
-            if (marker != JsonConstants.Asterisk)
-            {
-                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedStartOfValueNotFound, JsonConstants.Slash);
-            }
-
-            return SkipMultiLineCommentMultiSegment(localBuffer.Slice(1), leftOver);
-        }
+                ReadOnlySequence<byte> commentSequence = _sequence.Slice(start, end);
+                commentSequence = commentSequence.Slice(2, commentSequence.Length - 2 - tailBytesToIgnore);
```

Comment:
Can we try to avoid the double slicing of the `ReadOnlySequence`?
Maybe we can already have the start `SequencePosition` be off-set by 2. and change `end` to account for the `- 2 - tailBytesToIgnore`

Reply:
I don't believe this is possible. SequencePosition doesn't know how to move by 2 since it may end up in different object which it doesn't know about (where ReadOnlySequence does)